### PR TITLE
User locale update

### PR DESCRIPTION
**Pull request by mooreds**
Added some more info around locales and how FusionAuth works with them.

### CODE REVIEW

**Comment by mooreds** [comment]
Planning to merge this today or tomorrow @robotdan

File: site/docs/v1/tech/core-concepts/users.adoc [hunk 1]
```diff
@@ -112,3 +118,15 @@ It is possible, though rare, for an Elasticsearch index to become out of sync wi
...
+
+== User localization
+
+FusionAuth is built from the ground up with localization in mind. You can assign a user's `preferredLanguages` via the API or link:/docs/v1/tech/guides/advanced-registration-forms/[advanced registration form]. and add that value to a JWT or retrieve it from via the link:/docs/v1/tech/apis/users[User API]. Each application registration can also have a preferred language associated with it.
+
+Both the hosted login pages and the emails can be localized; the former via link:/docs/v1/tech/themes/localization/[themes] and the latter via link:/docs/v1/tech/email-templates/email-templates/[email templates].
+
+If you set the user's preferred language, FusionAuth's administrative user interface will localized. There are no translations of the FusionAuth administration user interface, however. 
+
+For instance, in the below screenshot, Richard's account has a `preferredLanguage` of `French`. When Richard interacts with the FusionAuth administrative user interface,  the account with the email dinesh@fusionauth.io, which was created on Oct 5, 2020, has a creation date display formatted as specified by the French locale, with the day first.
```
Comment (robotdan) [inline site/docs/v1/tech/core-concepts/users.adoc:130]: Not sure we need to mention this or not.. but we currently support the following locales as it relates to formatting date and time strings.  Where the suffix is the locale 

```
en
ar
cs
da
de
el
en
en_US
es
fr
ga
he
hi
it
ja
ko
nl
no
pt
ru
sv
zh
zh_CN
zh_TW
```
( https://github.com/FusionAuth/fusionauth-app/tree/691f9f196a390205d027aec284d164a91bb04085/web/WEB-INF/messages see `package.*` in this directory. 

These are easy to add, all we keep are the date and time format strings. For example, for French (`fr`) we have

```
date-format=d/M/yyyy
date-time-format=d/M/yyyy hh:mm a z
date-time-seconds-format=d/M/yyyy hh:mm:ss a z
```

So if anyone finds that their preferred locale is not producing the date and time formats as they expect they should open an issue and we'll add it.

**Review by robotdan — COMMENTED** [review]
Looks good!

**Comment by mooreds** [comment]
Sweet, I think this gives me enough stuff to make a top level core concept. Going to rework and have you re-review.

File: site/docs/v1/tech/core-concepts/localization-and-internationalization.adoc [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+* A locale identifies a language and a geographic region. An example is `fr_CA`, which means Canadian French, as opposed to French French, which is `fr_FR`. Java has https://www.oracle.com/technical-resources/articles/javase/locale.html[some good locale documentation].
+
+The key data element most relevant to l10n and i18n in FusionAuth is the [field]#preferredLanguages# field. This is present in both the User and Registration objects and stores a user's locale. The locale is available in the Registration because different applications may support different languages.
+
+There are four different areas of FusionAuth where user language preferences matter.
+
+* Your application
+* Hosted login pages
+* Emails and other messages
+* The administrative user interface
```
Comment (robotdan) [inline site/docs/v1/tech/core-concepts/localization-and-internationalization.adoc:23]: i.e. The FusionAuth Admin UI

File: site/docs/v1/tech/core-concepts/localization-and-internationalization.adoc [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+The default application registration form has the [field]#preferredLanguages# field as well.
+
+image::core-concepts/user-registration-locale.png[Adding a registration via the administrative interface allows you to set the preferred language of the user.,width=1200,role=bottom-cropped]
+
+After the user has a locale assigned, you can add it to a JWT using a link:/docs/v1/tech/lambdas/jwt-populate[JWT populate lambda], so it can be utilized by other applications.
+
+[source,javascript]
+----
+function populate(jwt, user, registration) {
+  jwt.preferredLanguages = registration.preferredLanguages || 'en_US';
```
Comment (robotdan) [inline site/docs/v1/tech/core-concepts/localization-and-internationalization.adoc:38]: Should this be as follows to preserve the data type with the default value?

```
jwt.preferredLanguages = registration.preferredLanguages || [ 'en_US' ]; 
```

**Review by robotdan — COMMENTED** [review]
Looks good

**Pull request merged**